mmdet/models/detectors/base.py

```diff
@@ -148,11 +146,14 @@ def forward(self, img, img_metas, return_loss=True, **kwargs):
         else:
             return self.forward_test(img, img_metas, **kwargs)
 
-    def show_result(self, data, result, dataset=None, score_thr=0.3):
-        if isinstance(result, tuple):
-            bbox_result, segm_result = result
-        else:
-            bbox_result, segm_result = result, None
+    def show_result(self,
+                    data,
+                    result,
+                    dataset=None,
+                    score_thr=0.3,
+                    show=False,
+                    images_out_dir=None):
+        from mmdet.apis.inference import show_result as show_single_result
```

Comment:
`mmdet.apis` contains high-level APIs exposed to basic users, and it should not be relied on by other modules. Different models may have different visualization methods. It would be better to call `detector.show_result()` in the inference API.

Reply:
I refactored `mmdet.apis` to use `detector.show_result` method.

I'm not sure if what I did was the right approach but, having in mind your point (Different models may have different visualization methods.), I:

- Made  `detector.show_result` expect a single image as input (instead of previous `data` dict). I could have added some conditionals and make it work with both single image and data but this felt cleaner. 
- Refactor single_gpu_test to adapt to the single image interface.